compiler/src/dmd/escape.d

```diff
@@ -2591,29 +2592,11 @@ private bool checkScopeVarAddr(VarDeclaration v, Expression e, Scope* sc, bool g
 /****************************
  * Determine if `v` is a typesafe variadic parameter.
  * Params:
- *      fd = function that `v` is in
  *      v = variable to check
  * Returns:
  *      true if `v` is a variadic parameter
  */
-bool isTypesafeVariadicParameter(FuncDeclaration fd, VarDeclaration v)
+bool isTypesafeVariadicParameter(VarDeclaration v)
 {
-    if (!(v.storage_class & STC.parameter))
-        return false;
-    if (auto tf = fd.type.toBasetype().isTypeFunction())
-    {
-        if (tf.parameterList.varargs != VarArg.typesafe)
-            return false;
-        if (const dim = tf.parameterList.length)
-        {
-            Parameter fparam = tf.parameterList[dim - 1];
-            if (fparam.ident == v.ident)
-            {
-                Type t = v.type.toBasetype();
-                if (t.isTypeDArray() || t.isTypeClass())
-                    return true;
-            }
-        }
-    }
-    return false;
+    return !!(v.storage_class & STC.variadic);
```

Comment:
Why the double negation?

Reply:
to not rely on the implicit conversion of `STC -> bool` presumably

Reply:
Because "cannot implicitly convert expression `v.storage_class & STC.variadic` of type `ulong` to `bool`"